theseus_gui/src/pages/project/Gallery.vue

```diff
@@ -0,0 +1,293 @@
+        <img :src="image.url" :alt="image.title" class="gallery-image" />
+      </a>
+      <div class="gallery-body">
+        <h3>{{ image.title }}</h3>
+        {{ image.description }}
+      </div>
+      <span class="gallery-time">
+        <CalendarIcon />
+        {{
+          new Date(image.created).toLocaleDateString('en-US', {
+            year: 'numeric',
+            month: 'long',
+            day: 'numeric',
+          })
+        }}
+      </span>
+    </Card>
+  </div>
+  <div v-if="expandedGalleryItem" class="expanded-image-modal" @click="expandedGalleryItem = null">
+    <div class="content">
+      <img
+        class="image"
+        :class="{ 'zoomed-in': zoomedIn }"
+        :src="
+          expandedGalleryItem.url
+            ? expandedGalleryItem.url
+            : 'https://cdn.modrinth.com/placeholder-banner.svg'
+        "
+        :alt="expandedGalleryItem.title ? expandedGalleryItem.title : 'gallery-image'"
+        @click.stop=""
+      />
+
+      <div class="floating" @click.stop="">
+        <div class="text">
+          <h2 v-if="expandedGalleryItem.title">
+            {{ expandedGalleryItem.title }}
+          </h2>
+          <p v-if="expandedGalleryItem.description">
+            {{ expandedGalleryItem.description }}
+          </p>
+        </div>
+        <div class="controls">
+          <div class="buttons">
+            <Button class="close" icon-only @click="expandedGalleryItem = null">
+              <XIcon aria-hidden="true" />
+            </Button>
+            <a
+              class="open btn icon-only"
+              target="_blank"
+              :href="
+                expandedGalleryItem.url
+                  ? expandedGalleryItem.url
+                  : 'https://cdn.modrinth.com/placeholder-banner.svg'
+              "
+            >
+              <ExternalIcon aria-hidden="true" />
+            </a>
+            <Button icon-only @click="zoomedIn = !zoomedIn">
+              <ExpandIcon v-if="!zoomedIn" aria-hidden="true" />
+              <ContractIcon v-else aria-hidden="true" />
+            </Button>
+            <Button v-if="gallery.length > 1" class="previous" icon-only @click="previousImage()">
+              <LeftArrowIcon aria-hidden="true" />
+            </Button>
+            <Button v-if="gallery.length > 1" class="next" icon-only @click="nextImage()">
+              <RightArrowIcon aria-hidden="true" />
+            </Button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import {
+  Card,
+  ExpandIcon,
+  RightArrowIcon,
+  LeftArrowIcon,
+  ExternalIcon,
+  ContractIcon,
+  XIcon,
+  CalendarIcon,
+  Button,
+} from 'omorphia'
+import { computed } from 'vue'
+
+const props = defineProps({
+  project: {
+    type: Object,
+    default: () => {},
+  },
+})
+
+const gallery = computed(() => props.project.gallery).value
+
+let expandedGalleryItem
+let expandedGalleryIndex
+let zoomedIn = false
```

Comment:
since these are used in the DOM they should be wrapped with refs. Not sure if it works ATM. ex:

`let expandedGalleryItem = ref(0)`
When you use it in `script setup`:
`expandedGalleryItem.value = 1`